### DISCUSSION TITLE
Milvus 2.0.0 Slow Query having few records on GPU Machine

### DISCUSSION THREAD

**Discussion by haresh1288** [milvus-io/milvus#15442] [Q&A and General discussion]
Hi, I am using Milvus 2.0.0 standalone on GPU Machine. 
I have inserted 16 Images Embedding Vector. Did the indexing and load before query or search operation. the search result is taking around 0.2 Second (only the collection.search line).

I did the same setup on CPU only machine & getting ~0.01 query execution time. Can you please provide me guidance that is there any GPU/CPU related configuration available on 2.0.0 Version? Or is there anything I have to do configuration on Milvus 2.0.0 ?

**Comment by yhmo** [comment]
The milvus has an internal timestamp machinery to sync data and operations, default is 200 ms. This time gap is too make sure all data is consumed before the search request is executed. You can specify a parameter "guarantee_timestamp" to change it:

```
search_param = {
    "data": search_vectors,
    "anns_field": vector_field,
    "param": {"metric_type": METRIC_TYPE, "params": {"nprobe": NPROBE}},
    "limit": TOPK,
    "expr": "id_field > 0"
    "guarantee_timestamp": 1}
results = collection.search(**search_param)
```

Set "guarantee_timestamp = 1" will tell milvus execute search request instantly.

**Reply by haresh1288** [reply]
Thank you @yhmo, let me check with the configuration.
